docs/en/rules/Azure.FrontDoor.Logs.md

```diff
@@ -25,6 +25,103 @@ Management operations for Front Door is captured automatically within Azure Acti
 
 Consider configuring diagnostics setting to log network activity through Front Door.
 
+## EXAMPLES
+
+<!-- 
+### Configure with Azure template
+
+To deploy Front Door that pass this rule:
+
+- Deploy a diagnostic settings sub-resource.
+- Enable logging for the `AuditEvent` category.
+
+For example:
+
+```json
+{
+    "coming soon once Bicep compiles and deploys...",
+
+}
+```
+-->
+
+---
+
+### Configure Manually
+
+To deploy Front Door that passes this rule:
+
+- Deploy a diagnostic settings sub-resource.
+- Manually go into the diagnostic settings and add these two rules:
+  - Enable logging for the `FrontdoorAccessLog` category.
+  - Enable logging for the `FrontdoorWebApplicationFirewallLog` category.
+
+---
+
+### Configure with Azure template
+
+If this setting was available in Bicep, it could be set like the following Bicep shows. 
+
+#### Bicep Template
+
+```bicep
+param frontDoorName string = 'myFrontDoor'
+param workSpaceId string = ''
+param location string = resourceGroup().location
+
+resource frontDoorResource 'Microsoft.Cdn/profiles@2021-06-01' = {
+  name: frontDoorName
+  location: 'Global'
+  sku: {
+    name: 'Standard_AzureFrontDoor'
+  }
+}
+
+resource frontDoorInsightsResource 'Microsoft.Cdn/profiles/providers/diagnosticSettings@2020-01-01-preview' = {
+  name: '${frontDoorName}/Microsoft.Insights/service'
+  dependsOn: [
+    frontDoorResource
+  ]
+  location: location
+  properties: {
+    workspaceId: workSpaceId
+    logs: [
+      {
+        category: 'FrontdoorAccessLog'
+        enabled: true
+      }
+      {
+        category: 'FrontdoorWebApplicationFirewallLog'
+        enabled: true
+      }
+    ]
+  }
+}
+```
+
+---
+
+#### Test Script
+
+Deploy Test Script: For debugging purposes, save the bicep above to sampleFrontDoor.bicep and run this command to test the deploy of this template:
+
+```ps1
+az deployment group create -n front-door-test-20220919T150200Z --resource-group rg_sandbox --template-file 'sampleFrontDoor.bicep' --parameters frontDoorName=yourFrontDoorName workSpaceId=/subscriptions/yourSubscriptionId/resourcegroups/yourResourceGroup/providers/microsoft.operationalinsights/workspaces/yourLogAnalyticsWorkspaceName
+```
+
+#### Errors
+
+However, this Bicep fails and gets the following error, so I can't figure out how to deploy this setting via ARM/Bicep.
+All the docs  say to do this manually... I can't find how to automate this setting!
```

Comment:
Try the suggestions above. A full sub-resource for key vault would look like:

```bicep
resource logs 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
  name: 'service'
  scope: vault
  properties: {
    workspaceId: workspaceId
    logs: [
      {
        category: 'AuditEvent'
        enabled: true
      }
    ]
  }
}
```

Reply:
I can get this to compile for the key vault, but the Front Door resource is scoped differently, so it throws an error with this syntax when I try to compile the bicep.

Reply:
@lluppesms Maybe try to be explicit in the default scope of the module.

Have a look at this full deployment for front door:

```bicep
// Create or update a Front Door
targetScope = 'resourceGroup'

// ----------
// PARAMETERS
// ----------

@description('The name of the Front Door.')
param name string = deployment().name

@allowed([
  'Premium'
  'Standard'
])
@description('The SKU of the Front Door.')
param sku string = 'Standard'

@description('Configure TLS certificates for custom domains.')
param certificates array = []

@metadata({
  strongType: 'Microsoft.OperationalInsights/workspaces'
  example: '/subscriptions/<subscription_id>/resourceGroups/<resource_group>/providers/Microsoft.OperationalInsights/workspaces/<workspace_name>'
})
@description('The resource ID to a Log Analytics workspace.')
param workspaceId string = ''

@description('Configures the log categories to send to the Log Analytics workspace.')
param diagnosticLogs array = [
  'FrontDoorAccessLog'
]

@description('Tags to apply to the resource.')
param tags object = {}

// ---------
// VARIABLES
// ---------

var allTags = union(resourceGroup().tags, tags)
var skuName = '${sku}_AzureFrontDoor'

// ---------
// RESOURCES
// ---------

@description('Create or update a Front Door.')
resource profile 'Microsoft.Cdn/profiles@2021-06-01' = {
  name: name
  location: 'Global'
  sku: {
    #disable-next-line BCP036
    name: skuName
  }
  properties: {
    originResponseTimeoutSeconds: 60
  }
  tags: allTags
}

@description('Configure diagnostics logs to send to a workspace.')
resource logs 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (!empty(workspaceId) && !empty(diagnosticLogs)) {
  name: 'logs'
  scope: profile
  properties: {
    workspaceId: workspaceId
    logs: [for item in diagnosticLogs: {
      category: item
      enabled: true
    }]
  }
}

@description('Configure TLS certificates for custom domains.')
resource certificate 'Microsoft.Cdn/profiles/secrets@2021-06-01' = [for item in certificates: {
  parent: profile
  name: '${split(item, '/')[8]}-${last(split(item, '/'))}'
  properties: {
    parameters: {
      type: 'CustomerCertificate'
      secretSource: {
        id: item
      }
      useLatestVersion: true
    }
  }
}]

// -------
// OUTPUTS
// -------

@description('A unique identifier for the Front Door.')
output id string = profile.id

@description('The name of the Resource Group where the Front Door is deployed.')
output resourceGroupName string = resourceGroup().name

@description('The guid for the subscription where the Front Door is deployed.')
output subscriptionId string = subscription().subscriptionId

@description('A unique guid for the Front Door included in forwarded headers.')
output frontDoorId string = profile.properties.frontDoorId

```